### ISSUE TITLE
Setting for ClientBufferMaxKBSize was non-int or non-positive. 

### ISSUE THREAD

**Issue by Flawm** [Microsoft/msphpsql#792]
### Description

We are running into a hiccup with ClientBufferMaxKBSize causing a total failure on every query when it's set to 0 in the php.ini.

###  Expected and actual behavior
As stated [here](https://docs.microsoft.com/en-us/sql/connect/php/sqlsrv-configure?view=sql-server-2017), when it is 0 there is supposed to be no buffer limit, however when we configure it to 0 in the php.ini, no queries work at all, and we get this from sqlsrv_errors()

```Array (
    [0] => Array ( [0] => IMSSP [SQLSTATE] => IMSSP [1] => -60 [code] => -60 [2] => Setting for ClientBufferMaxKBSize was non-int or non-positive. [message] => Setting for ClientBufferMaxKBSize was non-int or non-positive. ))
```
###  Steps to reproduce
Edit the php.ini to contain this
>sqlsrv.ClientBufferMaxKBSize = 0

and try to do any query.

After a cursory glance I want  to say this is being specifically thrown from L630 in source/sqlsrv/util.cpp but I really don't know what's going on here.
 
---
### PHP version
>php -v
>PHP 7.2.5 (cli) (built: Apr 25 2018 02:19:27) ( NTS MSVC15 (Visual C++ 2017) x64 )
>Copyright (c) 1997-2018 The PHP Group

### MSODBCSQL version
>17.1.0.1

### SQL Server version
>MSSQL Server 2014
>Microsoft SQL Server Management Studio						12.0.5207.0
>Microsoft Analysis Services Client Tools						12.0.5207.0
>Microsoft Data Access Components (MDAC)						6.3.9600.17415
>Microsoft MSXML						3.0 6.0 
>Microsoft Internet Explorer						9.11.9600.19003
>Microsoft .NET Framework				a		4.0.30319.42000
>Operating System						6.3.9600

### Client operating system
> Windows Server 2012 R2 Standard

### PHP SQLSRV extension version
>ExtensionVer | 5.1.0-preview+10119
>using (php_sqlsrv_72_nts_x64.dll)

**Comment by david-puglielli** [comment]
@Flawm The drivers don't allow ClientBufferMaxKBSize to be zero because 1. It causes problems with varchar(max) fields (see #228 and the fix #254), and 2. Buffered queries work best for small and medium sized result sets. The documentation is in error, for which we apologise. We will fix it promptly.

**Comment by Flawm** [comment]
Ok. Anyways, as a work around we're going to set it to the same value as our php memory limit.
